packages/input-select/src/Select.js

```diff
@@ -0,0 +1,159 @@
+    this._shouldItemRender = this._shouldItemRender.bind(this);
+    this.onFilter = this.onFilter.bind(this);
+    this.onSelect = this.onSelect.bind(this);
+  }
+
+  render() {
+    return (
+      <DropdownWrapper>
+        <ReactAutocomplete
+          items={this.props.items}
+          shouldItemRender={this._shouldItemRender}
+          getItemValue={item => item.label}
+          value={this.state.selectedLabel}
+          onChange={this.onFilter}
+          onSelect={this.onSelect}
+          renderInput={this._renderInput}
+          renderMenu={this._renderMenu}
+          renderItem={this._renderItem}
+          autoHighlight={false}
+          onMenuVisibilityChange={isMenuOpen => this.setState({ isMenuOpen })}
+        />
+      </DropdownWrapper>
+    );
+  }
+
+  onFilter(event) {
+    !this.state.isSearching && this.props.onChange("");
+
+    this.setState({ selectedLabel: event.target.value, isSearching: true });
+  }
+
+  onSelect(selectedLabel, item) {
+    this.setState({ selectedLabel, isSearching: false });
+    this.props.onChange(item.value);
+  }
+
+  getSelectedLabel(props) {
+    const item = props.value && props.items.find(x => x.value === props.value);
+
+    return item && item.label;
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (
+      newProps.value &&
+      newProps.value !== this.props.value &&
+      newProps.value !== this.state.selectedValue
+    ) {
+      this.setState({
+        selectedValue: newProps.value,
+        selectedLabel: this.getSelectedLabel(newProps)
+      });
+    }
+  }
+
+  _renderInput(autoCompleteProps) {
+    const { ref, ...rest } = autoCompleteProps;
+
+    const { renderItem, ...inputProps } = this.props;
+
+    inputProps.validationMessages = this.state.isMenuOpen
+      ? []
+      : this.props.validationMessages;
+
+    return <EnhancedInput {...inputProps} {...rest} innerRef={ref} />;
+  }
+
+  _renderMenu(items) {
+    if (!items || !items.length) {
+      return <EmptyCard noResultsMessage={this.props.noResultsMessage} />;
+    }
+
+    return (
+      <DropdownMenuWrapper>
+        <ul>{items}</ul>
+      </DropdownMenuWrapper>
+    );
+  }
+
+  _renderItem(item, highlighted) {
+    return (
+      <DropdownItemWrapper
+        key={item.value}
+        highlighted={highlighted}
+        selected={this.state.selectedLabel === item.label}
+      >
+        <div>
+          {this.props.renderItem ? this.props.renderItem(item) : item.label}
+        </div>
+      </DropdownItemWrapper>
+    );
+  }
+
+  _shouldItemRender(item) {
+    if (this.state.isSearching) {
+      return (
+        item.label
+          .toLowerCase()
+          .indexOf(this.state.selectedLabel.toLowerCase()) > -1
+      );
+    }
+
+    // If user is not searching, we render the full list of items
+    return true;
+  }
+}
+
+Select.defaultProps = {
+  onChange: () => false
+};
+
+Select.propTypes = {
+  items: PropTypes.arrayOf(
+    PropTypes.shape({
```

Comment:
I dont know about this fixed shape here, if the renderItem is used, items can have any shape

Reply:
Sure... Did it this way to have a pattern when we use this component, if needed we can change this in the future for sure.